homeassistant/components/pushbullet/sensor.py

```diff
@@ -75,92 +91,123 @@
+    async_create_issue(
+        hass,
+        DOMAIN,
+        "deprecated_yaml",
+        breaks_in_ha_version="2023.1.0",
+        is_fixable=False,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Pushbullet sensors from config entry."""
 
-    try:
-        pushbullet = PushBullet(config.get(CONF_API_KEY))
-    except InvalidKeyError:
-        _LOGGER.error("Wrong API key for Pushbullet supplied")
-        return
-
-    pbprovider = PushBulletNotificationProvider(pushbullet)
-
-    monitored_conditions = config[CONF_MONITORED_CONDITIONS]
+    pushbullet: PushBullet = hass.data[DOMAIN][entry.entry_id]
+    pb_provider = PushBulletNotificationProvider(hass, pushbullet)
     entities = [
-        PushBulletNotificationSensor(pbprovider, description)
+        PushBulletNotificationSensor(hass, entry, pb_provider, description)
         for description in SENSOR_TYPES
-        if description.key in monitored_conditions
     ]
-    add_entities(entities)
+
+    async_add_entities(entities)
 
 
 class PushBulletNotificationSensor(SensorEntity):
     """Representation of a Pushbullet Sensor."""
 
+    _attr_should_poll = False
+    _attr_has_entity_name = True
+
     def __init__(
         self,
-        pb,  # pylint: disable=invalid-name
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        pb_provider: PushBulletNotificationProvider,
         description: SensorEntityDescription,
-    ):
+    ) -> None:
         """Initialize the Pushbullet sensor."""
+        self.hass = hass
         self.entity_description = description
-        self.pushbullet = pb
-
-        self._attr_name = f"Pushbullet {description.key}"
-
-    def update(self) -> None:
+        self.pb_provider = pb_provider
+        self._attr_unique_id = f"{entry.entry_id}-{description.key}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.pb_provider.pushbullet.user_info["iden"])},
+            name=entry.data[CONF_NAME],
+            entry_type=DeviceEntryType.SERVICE,
+        )
+
+    @callback
+    def async_update_callback(self) -> None:
         """Fetch the latest data from the sensor.
 
         This will fetch the 'sensor reading' into self._state but also all
         attributes into self._state_attributes.
         """
         try:
-            self._attr_native_value = self.pushbullet.data[self.entity_description.key]
-            self._attr_extra_state_attributes = self.pushbullet.data
+            self._attr_native_value = self.pb_provider.data[self.entity_description.key]
+            self._attr_extra_state_attributes = self.pb_provider.data
```

Comment:
It's seems we're duplicating the state in the extra state attributes. We normally don't allow that.

Reply:
I think we should drop the extra attributes as they are available as sensor entities. This was ported from the existing code so removing it would be a breaking change. Should we remove it?

Reply:
Yeah, I think so, either here or in a follow up PR.

Reply:
I suggest to keep it as it is in this PR and remove in a follow up PR.